tcms/testplans/static/testplans/js/get.js

```diff
@@ -19,36 +24,45 @@ $(document).ready(function() {
                 testCases.not_confirmed.push(data[i]);
             }
         }
-        drawTestCases(testCases.confirmed);
+        drawTestCases(testCases.confirmed, testPlanId, permissions);
         treeViewBind();
     });
 
 });
 
-function drawTestCases(testCases) {
+function drawTestCases(testCases, testPlanId, permissions) {
     var container = $('#confirmed-testcases'),
         noCasesTemplate = $('#no_test_cases'),
         testCaseRowDocumentFragment = $('#test_case_row')[0].content;
 
     if (testCases.length > 0) {
         for (var i = 0; i < testCases.length; i++) {
-            container.append(getTestCaseRowContent(testCaseRowDocumentFragment.cloneNode(true), testCases[i]));
+            container.append(getTestCaseRowContent(testCaseRowDocumentFragment.cloneNode(true), testCases[i], permissions));
+        }
+        attachEvents(testPlanId, permissions);
+        if (permissions['perm-change-testcase']) {
+            drawStatusAndPriority();
         }
     } else {
         container.append(noCasesTemplate[0].innerHTML);
     }
 }
 
 
-function getTestCaseRowContent(rowContent, testCase) {
+function getTestCaseRowContent(rowContent, testCase, permissions) {
     var row = $(rowContent);
 
-    row.find('.js-test-case-link').html(`TC-${testCase.id}: ${testCase.summary}`).attr('href',`/case/${testCase.id}/`);
+    row[0].firstElementChild.dataset.testcasePk = testCase.id;
+    row.find('.js-test-case-link').html(`TC-${testCase.id}: ${testCase.summary}`).attr('href', `/case/${testCase.id}/`);
     row.find('.js-test-case-priority').html(`${testCase.priority}`);
     row.find('.js-test-case-category').html(`${testCase.category}`);
     row.find('.js-test-case-author').html(`${testCase.author}`);
     row.find('.js-test-case-tester').html(`${testCase.default_tester || '-'}`);
 
+    // set the link in the kebab menu
+    if (permissions['perm-change-testcase']) {
```

Comment:
If the user doesn't have the change permission the HTML will not render the `.js-test-case-menu-edit`!  I'm not sure if you should be duplicating the permission check everywhere or instead make sure the underlying element actually exists before manipulating it.

Reply:
Yes it will not render. The next row would fail if we miss the check.Also its faster to check up the key in the permissions then diving into the DOM.